### DISCUSSION TITLE
local anywidget failure when plotting polygons but works fine in binder

### DISCUSSION THREAD

**Discussion by emileten** [developmentseed/lonboard#225] [Q&A] [answered]
I am trying to use `lonboard.SolidPolygonLayer` to plot polygons from a list of GeoJSONs that I load using geopandas. The rendering fails with what seems to be an `anywidget` error (see screen below, plus console screen). 

Crucially, the exact same task runs fine if I run it in the binder environment provided in this repository, so it has something to do with my set up. I tried to replicate the binder environment by pip-freezing it and wrapping it all up in a conda environment with the same python version as in binder, but the error remains. 

Any thoughts ? Any help is welcome, thanks for reading this (And for the nice package 😄) 

Here is the information necessary to reproduce the error.

- [notebook to run.](https://gist.github.com/emileten/aa24ddac2d2b413672250f797c498499)

- the [json file with an example of the items](https://gist.github.com/emileten/d45bbc07a269ac3c82e83aa16ea11218) of which I am trying to plot the boundaries (this json file is read in the notebook)

- To mimic my environment, run the notebook in a conda environment [using this configuration file.](https://gist.github.com/emileten/de9dac79a3e940b183316d3352367370) This configuration is meant to mimic the binder environment provided in this repository : all packages, including jupyter related ones, are the same version. 

![Screenshot 2023-11-09 at 8 27 31 PM](https://github.com/developmentseed/lonboard/assets/45140658/f70fd96f-e1ef-43e0-8501-bb25ce08f32f)

![Screenshot 2023-11-09 at 8 29 08 PM](https://github.com/developmentseed/lonboard/assets/45140658/fdce509d-3191-469f-ac13-7f07915be08d)

**Comment by kylebarron** [comment]
`SyntaxError: invalid regular expression flag` usually means that the JS part of lonboard wasn't included in the install.

How did you install it?

> run the notebook in a conda environment [using this configuration file.](https://gist.github.com/emileten/de9dac79a3e940b183316d3352367370)

Ah looks like lonboard is incorrectly packaged for conda, cc @giswqs in https://github.com/developmentseed/lonboard/pull/223

**Reply by emileten** [reply]
@kylebarron I install it _through pip within conda_ (see the config file) so this shouldn't make a difference ?

Can you share how you are installing it (e.g just virtual env and pip install lonboard?), so that I can try to replicate ?

Thanks !

**Reply by kylebarron** [reply]
Oh I see. Can you try reloading the browser page? The only time I've seen that error is where the JS part wasn't included.

I usually just create a virtualenv and install lonboard in it. It probably makes sense to add a `troubleshooting` doc to the website

**Reply by emileten** [reply]
Another approach resulted in the same error

```
git clone git@github.com:developmentseed/lonboard.git
cd lonboard
python -m venv .venv
source .venv/bin/activate
pip install -e .
pip install ipykernel jupyter jupyterlab
python -m ipykernel install --user --name=lonboard-venv
jupyter lab
```

Then running the above notebook (also tried reloading the browser page)

**Reply by kylebarron** [reply]
you can't install from git without manually building the js parts, see DEVELOP.md. Otherwise you need to pip install from PyPI, which will download a pre-built JS bundle as part of the python package

**Reply by vincentsarago** [reply]
@kylebarron do you think we could update the develop.md to make it a bit more clear what are the steps for local install? 🙏

**Reply by vincentsarago** [reply]
Oo @emileten started a PR do add some info 🙌

**Reply by emileten** [reply]
> you can't install from git without manually building the js parts, see DEVELOP.md

Yes thanks 👍.

> update the develop.md to make it a bit more clear what are the steps for local install?

that would be great ! I started a PR, but maybe pointing to develop.md in the main docs and make the JS build dependency more explicit in develop.md would be useful!

**Comment by emileten** [comment, marked as answer]
> The only time I've seen that error is where the JS part wasn't included.

Alright so, that's the answer to this problem, the JS code was not built (i.e I was missing the `lonboard/static` folder). I ran `npm run build` and my problem is fixed. The reason I didn't have the JS files in the first place is that I installed from source (pip install from cloned repository). I opened a PR to add in the docs an installation-from-source instruction https://github.com/developmentseed/lonboard/pull/226.
